### PR TITLE
Implement Numeric IDs migration

### PR DESCRIPTION
**Pull request by aerosol**
### Changes

First pass on Numeric IDs Clickhouse migration.
For now the PR contains the necessary SQL files + a coordination script.
I expect things to change going forward, but submitting this as an interim PR - no action required yet, should be completely transparent.

I am trying to work on an integration test on the side but it's currently painful due to clickhouse container having to call the host in order to retrieve domains lookup dictionary via HTTP. Awaiting clarity on clickhouse-postgres connectivity for that.

### Tests
- [ ] Automated tests have been added
- [x] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog
- [x] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [x] This change does not need a documentation update

### Dark mode
- [ ] The UI has been tested both in dark and light mode
- [x] This PR does not change the UI

### CODE REVIEW

**Comment by aerosol** [comment]
Rebased & adjusted for #2733

File: lib/plausible/data_migration/numeric_ids.ex [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        Application.get_env(:plausible, Plausible.IngestRepo)[:url]
+      )
+
+    max_threads =
+      "NUMERIC_IDS_MIGRATION_MAX_THREADS" |> System.get_env("16") |> String.to_integer()
+
+    # TBD: There's most likely a bug in Clickhouse defining Postgres dictionaries,
+    # we'll use a static URL for now
+    dict_url = Keyword.get(opts, :dict_url) || System.get_env("DOMAINS_DICT_URL") || ""
+    dict_password = Keyword.get(opts, :dict_url) || System.get_env("DOMAINS_DICT_PASSWORD") || ""
```
Comment (vinibrsl) [inline lib/plausible/data_migration/numeric_ids.ex:26]: This is fetching the `:dict_url` from `opts` in both variables. Should we change the second to `dict_password`? Or is the URL already authenticated somehow?

**Review by vinibrsl — APPROVED** [review]
Looks good! Awesome work, @aerosol.

File: lib/plausible/data_migration/numeric_ids.ex (hunk 1 above)
Comment (aerosol) [reply lib/plausible/data_migration/numeric_ids.ex]: For our case the pass is already kept in the vault and provided via env vars, but good call, it's an omission on my behalf, thanks! 2823261f

**Pull request merged**